pkg/serverless/invocationlifecycle/init.go

```diff
@@ -150,19 +162,26 @@ func (lp *LifecycleProcessor) initFromSQSEvent(event events.SQSEvent) {
 	lp.addTag(tagFunctionTriggerEventSource, sqs)
 	lp.addTag(tagFunctionTriggerEventSourceArn, trigger.ExtractSQSEventARN(event))
 
-	// test for SNS
+	// Check for SNS event wrapped by the SQS body
 	var snsEntity events.SNSEntity
-	if err := json.Unmarshal([]byte(event.Records[0].Body), &snsEntity); err != nil {
-		return
+	if err := json.Unmarshal([]byte(event.Records[0].Body), &snsEntity); err == nil {
+		if strings.ToLower(snsEntity.Type) == "notification" && snsEntity.TopicArn != "" {
+			lp.createSNSInferredSpan(snsEntity)
+			return
+		}
 	}
```

Comment:
This already existing code added support for SNS -> SQS -> Lambda inferred spans